include/exadg/convection_diffusion/user_interface/parameters.cpp

```diff
@@ -256,6 +257,28 @@ Parameters::check() const
   // SPATIAL DISCRETIZATION
   grid.check();
 
+  if(enable_adaptivity)
+  {
+    AssertThrow(not ale_formulation,
+                dealii::ExcMessage("Combination of adaptive mesh refinement "
+                                   "and ALE formulation not implemented."));
+
+    AssertThrow(temporal_discretization == TemporalDiscretization::BDF,
+                dealii::ExcMessage("Combination of adaptive mesh refinement "
+                                   "and explicit time integration not implemented."));
+
+    AssertThrow(use_cell_based_face_loops == false,
+                dealii::ExcMessage("Cell-based face loops not supported for "
+                                   "adaptively refined meshes."));
+
+    if(involves_h_multigrid())
+    {
+      AssertThrow(grid.create_coarse_triangulations,
+                  dealii::ExcMessage("Coarse triangulations required for locally refined meshes"
+                                     "and h-multigrid."));
+    }
```

Comment:
Can we write this at a more central place? The idea behind is that we do not want to write such asserts for every module for which AMR is introduced. How about doing this check inside the `MultigridPreconditionerBase` class?

Reply:
In `MultigridPreconditionerBase`, we do not have the parameters readily available to check these conditions. But I think `GridUtilities::setup_after_coarsening_and_refinement()` might be well suited! I asserted simplex elements and the condition we checked before in parameters. This achieves the point regarding application independency, as well. Please check
https://github.com/exadg/exadg/pull/586/commits/bbfad6f3df4cdaab742ff80c8e9e584a1e08c272

Reply:
Let's discuss this also in person and see which options we have.